packages/react/src/Avatar/Avatar.tsx

```diff
@@ -0,0 +1,47 @@
+import React, {forwardRef, useMemo} from 'react'
+import clsx from 'clsx'
+import styles from './Avatar.module.css'
+
+import type {BaseProps} from '../component-helpers'
+
+export const AvatarSizes = [32, 40, 48, 64, 80] as const
+export const AvatarShapes = ['circle', 'square'] as const
+
+type AvatarSizeVariants = typeof AvatarSizes[number]
+
+type ResponsiveSizeMap = {
+  narrow?: AvatarSizeVariants
+  regular?: AvatarSizeVariants
+  wide?: AvatarSizeVariants
+}
+
+export type AvatarProps = BaseProps<HTMLSpanElement> & {
+  size?: typeof AvatarSizes[number] | ResponsiveSizeMap
+  shape?: typeof AvatarShapes[number]
+  src: string
+  alt: string
+} & React.HTMLAttributes<HTMLSpanElement>
+
+export const Avatar = forwardRef<HTMLSpanElement, AvatarProps>(
+  ({alt, className, size = AvatarSizes[1], shape = AvatarShapes[0], src, ...props}, ref) => {
+    const sizeClass = useMemo(() => {
+      return typeof size === 'number'
+        ? styles[`Avatar--size-${size}`]
+        : Object.keys(size)
+            .map(viewport => {
+              return styles[`Avatar-${viewport}--size-${size[viewport]}`]
+            })
+            .join(' ')
+    }, [size])
+
+    return (
+      <span
+        ref={ref}
+        className={clsx(styles.Avatar, sizeClass, styles[`Avatar--shape-${shape}`], className)}
```

Comment:
What do you think about forwarding these classes to the `img` instead? If a user passes a custom `className`, would they expect it to apply to the image or root `span`? They may also need to forward some other attributes, like image width and height. What do you think @danielguillan?

Reply:
I'm not sure about doing this. We generally forward `classname` to the root element on most of our components. Also, any style overrides must target the root element to work correctly (thinking about adding a border style, for example). Finally, as the image element is set to `object-fit: cover`, passing `width` and `height` attributes wouldn't have any effect.

Reply:
Okay 👍 .. flagging as we had / are having the same issue with CTABanner, where I'm needing to use `>` selector to target the inner element. As long as the image is customisable enough app side. I'll test it during release though. From a user-perspective, I still think I'd expect any props to be forwarded to the image not a wrapper component, which is purely implementation detail that's abstraction for the user.

Reply:
OK, I'm now forwarding props to the image element.